prov/rxm/src/rxm.h

```diff
@@ -572,7 +579,26 @@ rxm_process_recv_entry(struct rxm_recv_queue *recv_queue,
 		dlist_remove(&rx_buf->unexp_msg.entry);
 		rx_buf->recv_entry = recv_entry;
 		recv_queue->rxm_ep->res_fastlock_release(&recv_queue->lock);
-		return rxm_cq_handle_rx_buf(rx_buf);
+		if (rx_buf->pkt.ctrl_hdr.type != ofi_ctrl_seg_data) {
+			return rxm_cq_handle_rx_buf(rx_buf);
+		} else {
+			int last = !rx_buf->pkt.ctrl_hdr.seg_no;
```

Comment:
I have the same question as before.  I don't understand why last = !seg_no (or last = (seg_no == 0)).  That is unexpected.

If we can't figure out if a segment isn't the last based on its actual count, please use a LAST_SEGMENT_FLAG flag or set seg_no = LAST_SEGMENT define that we can check for.  Starting seg_no at 1, and using 0 to mark the last one is odd.

Reply:
to avoid carrying any additional flags in the pkt.hdr or last segment number.
For instance, there is large message [___(N-1)___|___(N-2)___| ... |___(1)___|___(0)___] that is splitted into N segments.
The segment is sent in the following order
- |___(N-1)___|
- |___(N-2)___|
...
- |___(1)___|
- |___(0)___|

So, the 0 is the last segment on a receiver side

Defining `RXM_SAR_LAST_SEGMENT` to 0 looks reasonable